src/trajectory.jl

```diff
@@ -563,22 +501,22 @@ function combine(treeleft::BinaryTree, treeright::BinaryTree)
     return BinaryTree(
         treeleft.zleft,
         treeright.zright,
-        combine(treeleft.c, treeright.c),
+        rho_merge(treeleft.rho, treeright.rho),
         treeleft.sum_α + treeright.sum_α,
         treeleft.nα + treeright.nα,
         maxabs(treeleft.ΔH_max, treeright.ΔH_max),
     )
 end
 
 """
-    isterminated(h::Hamiltonian, t::BinaryTree{<:ClassicNoUTurn})
+    isterminated(h::Hamiltonian, t::BinaryTree)
 
 Detect U turn for two phase points (`zleft` and `zright`) under given Hamiltonian `h`
 using the (original) no-U-turn cirterion.
 
 Ref: https://arxiv.org/abs/1111.4246, https://arxiv.org/abs/1701.02434
 """
-function isterminated(h::Hamiltonian, t::BinaryTree{<:ClassicNoUTurn})
+function isterminated(::ClassicNoUTurn, h::Hamiltonian, t::BinaryTree)
```

Comment:
After introducing a parametric turn statistic, this function can be simplified into 


```julia

function isterminated(left::TurnStatistic{ClassicNoUTurn}, right::TurnStatistic{ClassicNoUTurn}, h::Hamiltonian)

```

Reply:
I quite like this suggestion. Do we want to implement in this PR or a seperate though? It seems need some careful cares.